Expand LRA to compute also other statistics and some other refinements

jhardenberg opened the pull request:
## PR description:

This expands LRA capabilities to compute also additional statistics, not only the mean.
It contains also some other minor improvements to LRA functionality, in particular grouping annual files using cdo (which is significantly faster than using xarray). 

## Issues closed by this pull request:

Close #1885

----

 - [x] Tests are included if a new feature is included.
 - [x] Documentation is included if a new feature is included.
 - [x] Docstrings are updated if needed.
 - [x] Changelog is updated.

jhardenberg commented:
The first addition is a new `compact` keyword to `LRAgenerator` which allows specifying `cdo` as an alternative method to create the yearly files. In my tests trying to extract data for a limited area at 0.05 degrees, it was taking 2 minutes to generate each month and the above 50min to compact into a yearly file! Now with cdo this takes only one and half minutes. The options for cdo are by default to create a nc4 zipped level 1 file, but they are exposed with another keyword. To be decided if not to make 'cdo' the default option. Currently it is possible to choose 'cdo', 'xarray' or None (which would avoid running the compacting of the monthly in yearly files.

jhardenberg commented:
next change is the 'stat' keyword  (default is 'mean') . The output filenames are like "2t_ssp370_r005_daily_min_myregion_202001.nc"

jhardenberg commented:
One important point: since this chanegs the output naming, if this PR  is merged it will trigger a complete rebuild of all LRAs we have ...maybe we should omit 'mean' in the output filenames if the statistic is 'mean' ?  Or maybe we should rename by hand all output files before the next time LRA is run?

jhardenberg commented:
The new option to drop empty gridpoints for a regional selection is called 'drop' and the one to decide if and how to do the yearly compacting of files is called 'compact' .... suggestions welcome if you have better names

mnurisso commented:
Hi, I think we need to discuss this in the next in person meeting in order to be sure we do not only patch the LRA but we have a more organic update of the structure accomodating all the features. I'd like to have also the level selection from #1879 in this list of new features.

I'd also like to have both the possibility of having the kwargs as cli args but also in the yaml file. Maybe we need to think which structure this file should have to accommodate the features.

jhardenberg commented:
Ok, last changes which we agreed: 
* I implemented the new way to do 'drop' in the regional area selection. Tested in a workflow and works like a charm, producing nice and smaller files only for the region of interest. Possibly because it takes less to write the also now it takes just 1 minute to process one month of climatedt1 ICON ssp370 data regridded at 0.05 degrees
* removed the 'cdo' option but left 'cdo_options' (we will think again about this when we decide how to further change LRA)

oloapinivad commented on src/aqua/lra_generator/lra_generator.py:
```diff
@@ -39,11 +40,15 @@ def __init__(self,
+                 stat="mean",
+                 compact="xarray",
+                 cdo_options=["-f", "nc4", "-z", "zip_1"],
+                 **kwargs):
```
what is `zip_1`? why not simply a`zip`?

jhardenberg replied:
zip_1 is zip compression level 1
We use the same option when writing with xarray

oloapinivad replied:
I suspect it is the same of using `zip`! but of course no hassle, go on and merge for me

jhardenberg commented:
All works, tests passing, merging
